src/qt/guiutil.cpp

```diff
@@ -164,7 +164,7 @@ bool parseBitcoinURI(const QString &scheme, const QUrl &uri,
             fShouldReturnFalse = false;
         } else if (i->first == "amount") {
             if (!i->second.isEmpty()) {
-                if (!BitcoinUnits::parse(BitcoinUnits::BCH, i->second,
+                if (!BitcoinUnits::parse(BitcoinUnits::SAT, i->second,
```

Comment:
```suggestion
                if (!BitcoinUnits::parse(BitcoinUnits::LOTUS, i->second,
```
???

Reply:
Per the current spec, query parameters must use sats. We can change it if you want, but it results in confusion because of differences between wallets.

Reply:
so the current Bitcoin-QT impl is wrong?

Reply:
Current Bitcoin-QT implementation is correct given the lack of any specification of how the amount field should be handled within the current two address specifications. For Lotus, I'm specifying that it should always be sats. We could detect a decimal point and toggle between the two. That's what I had to do in Cashew -- but I really dislike it.

Reply:
I see.